cocos/animation/core/transform.ts

```diff
@@ -0,0 +1,228 @@
+import { Vec3 } from '../../core/math/vec3';
+import { Quat } from '../../core/math/quat';
+import { EPSILON, Mat4 } from '../../core';
+
+const CACHE_VECTOR_A = new Vec3();
+const CACHE_VECTOR_B = new Vec3();
+const CACHE_QUAT_A = new Quat();
+const CACHE_QUAT_B = new Quat();
+
+// Can not use `Readonly<Transform>`.
+// See: https://github.com/microsoft/TypeScript/issues/50668
+type ReadonlyTransform = Transform;
+
+export class Transform {
+    public static IDENTITY = Object.freeze(new Transform());
+
+    public static ZERO = Object.freeze((() => {
+        const transform = new Transform();
+        Vec3.copy(transform._position, Vec3.ZERO);
+        Quat.set(transform._rotation, 0.0, 0.0, 0.0, 0.0);
+        Vec3.copy(transform._scale, Vec3.ZERO);
+        return transform;
+    })());
+
+    get position (): Readonly<Vec3> {
+        return this._position;
+    }
+
+    set position (value) {
+        Vec3.copy(this._position, value);
```

Comment:
I think use `this._position.set(value)` will get better readability.

Reply:
[`Vec3.set`](https://github.com/cocos/cocos-engine/blob/v3.7.0/cocos/core/math/vec3.ts#L84) does not take a vector parameter.